### PR TITLE
[SEARCH-421, SEARCH-422] Extend JS tests with InvertedIndex and nested documents

### PR DESCRIPTION
**Pull request by alexbakharew**
### Scope & Purpose
[SEARCH-421](https://arangodb.atlassian.net/browse/SEARCH-421): Add documents with nested fields and index them as an addition for existing test cases. We will check that old queries work in the same way as it was before indexing nested fields.

[SEARCH-422](https://arangodb.atlassian.net/browse/SEARCH-422): If possible, add filter queries which will utilize inverted index. Results for arangosearch, search-alias and inverted index should be the same  

- [ ] :hankey: Bugfix
- [X] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [X] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [X] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1180
- [X] GitHub issue / Jira ticket: 
https://arangodb.atlassian.net/browse/SEARCH-421
https://arangodb.atlassian.net/browse/SEARCH-422
- [ ] Design document: 

[SEARCH-421]: https://arangodb.atlassian.net/browse/SEARCH-421?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SEARCH-422]: https://arangodb.atlassian.net/browse/SEARCH-422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alexbakharew** [comment]
Observed one known failure:
```
=== shell_client_0 ===
* Test "shell_client_0"
    [FAILED]  tests/js/common/shell/test-kill-queries.js

      "test_nonstream_directKillBeforeQueryResultIsGettingHandled" failed: Error: at assertion #1: assertEqual: (1500) is not equal to (1457)
(Error
    at assertEqual (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:108:19)
    at executeDefaultCursorQuery (tests/js/common/shell/test-kill-queries.js:55:7)
    at tests/js/common/shell/test-kill-queries.js:240:11
    at Object.run (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:552:27)
    at Object.Run [as run] (/work/ArangoDB/js/common/modules/jsunity.js:299:24)
    at tests/js/common/shell/test-kill-queries.js:281:11
    at tests/js/common/shell/test-kill-queries.js:285:3
    at RunTest (/work/ArangoDB/js/common/modules/jsunity.js:418:12)
    at testFunc (eval at runOneTest (/work/ArangoDB/js/client/modules/@arangodb/testutils/test-utils.js:501:7), <anonymous>:4:8)
    at runLocalInArangoshRunner.runOneTest (/work/ArangoDB/js/client/modules/@arangodb/testutils/test-utils.js:515:20))
    at Object.run (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:575:23)
    at Object.Run [as run] (/work/ArangoDB/js/common/modules/jsunity.js:299:24)
    at tests/js/common/shell/test-kill-queries.js:281:11
    at tests/js/common/shell/test-kill-queries.js:285:3
    at RunTest (/work/ArangoDB/js/common/modules/jsunity.js:418:12)
    at testFunc (eval at runOneTest (/work/ArangoDB/js/client/modules/@arangodb/testutils/test-utils.js:501:7), <anonymous>:4:8)
    at runLocalInArangoshRunner.runOneTest (/work/ArangoDB/js/client/modules/@arangodb/testutils/test-utils.js:515:20)
    at runLocalInArangoshRunner.run (/work/ArangoDB/js/client/modules/@arangodb/testutils/testrunner.js:523:28)
    at Object.shellClient [as shell_client] (/work/ArangoDB/js/client/modules/@arangodb/testsuites/aql.js:208:79)
    at iterateTests (/work/ArangoDB/js/client/modules/@arangodb/testutils/testing.js:522:36) - test failed

File: tests/js/client/communication/test-communication.js [hunk 1]
```diff
@@ -340,15 +348,19 @@ function GenericAqlSetupPathSuite(type) {
...
+        return `db._query("FOR v IN ${vertexName} FOR t IN 1 OUTBOUND v GRAPH ${graphName} FOR x IN 1..${docsPerWrite} INSERT {b: [{c: [{d: 'f'}]}]} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
       case "View":
-        return `db._query("FOR v IN ${viewName} OPTIONS {waitForSync: true} FOR x IN 1..${docsPerWrite} INSERT {} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
+        return `db._query("FOR v IN ${viewName} OPTIONS {waitForSync: true} FOR x IN 1..${docsPerWrite} INSERT {b: [{c: [{d: 'q'}]}]} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
+      case "SearchAliasView":
+        return `db._query("FOR v IN ${searchAliasViewName} OPTIONS {waitForSync: true} FOR x IN 1..${docsPerWrite} INSERT {b: [{c: [{d: 'q'}]}]} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
+      case "InvertedIndex":
+        return `db._query("FOR v IN ${vertexName} OPTIONS {waitForSync: true} FOR x IN 1..${docsPerWrite} INSERT {b: [{c: [{d: 'q'}]}]} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
```
Comment (MBkkt) [inline tests/js/client/communication/test-communication.js:361]: same here

File: tests/js/server/resilience/move-view/moving-shards-with-arangosearch-view-cluster.js_DISABLED [hunk 2]
```diff
@@ -608,19 +609,31 @@ function MovingShardsWithViewSuite (options) {
...
+            viewMeta = {links:
               {[name]:
                 {fields:
-                  {'a': { analyzers: ['identity']}, 'b': { analyzers: ['text_en']}}
+                  {'a': { analyzers: ['identity']}, 'b': { analyzers: ['text_en']}, "name_1": {}, "value": { "nested": { "nested_1": {"nested": {"nested_2": {}}}}}}
                 }
               }
-            });
+            };
+          } else {
```
[the remark this answers is not in the document]
Comment (alexbakharew) [reply tests/js/server/resilience/move-view/moving-shards-with-arangosearch-view-cluster.js_DISABLED:621]: I enabled this test and checked that my changes are working.

File: tests/js/server/aql/aql-geo.js [hunk 3]
```diff
@@ -141,26 +142,51 @@ function geoSuite(isSearchAlias) {
   }
 
   function compare(query, queryView) {
```
[the remark this answers is not in the document]
Comment (alexbakharew) [reply tests/js/server/aql/aql-geo.js:144]: Decided just leave everything as is

File: tests/js/server/aql/aql-geo.js [hunk 4]
```diff
@@ -182,20 +208,68 @@ function geoSuite(isSearchAlias, analyzerType) {
       withIndex.ensureIndex({type: "geo", geoJson: true, fields: ["geo"]});
       withView = db._create(collWithView);
       let analyzers = require("@arangodb/analyzers");
+      let a = analyzers.save("geo_json", "geojson", {}, ["frequency", "norm", "position"]);
```
Comment (MBkkt) [inline tests/js/server/aql/aql-geo.js:211]: Instead of "geojson" was `anaylyzerType`

**Review by MBkkt — APPROVED** [review]

File: tests/js/common/aql/aql-view-arangosearch-ddl-combined.js [hunk 5]
```diff
@@ -0,0 +1,2352 @@
...
+      }
+      db._createView("FOO_view", "arangosearch", meta1);
+      db._createView("FOO_view2", "arangosearch", meta2);
+
+      db._useDatabase("_system");
+      db._dropDatabase(databaseNameAnalyzer);
+      db._dropDatabase(databaseNameView);
+      analyzers.remove(systemAnalyzerName, true);
+    },
+    // Commented as this situation is not documented (user should not do this), will be adressed later
```
Comment (MBkkt) [inline tests/js/common/aql/aql-view-arangosearch-ddl-combined.js:1589]: Task or at least addressed todo?

File: tests/js/server/aql/aql-geo.js (hunk 4 above)
Comment (alexbakharew) [reply tests/js/server/aql/aql-geo.js]: Fixed

File: tests/js/common/aql/aql-view-arangosearch-ddl-combined.js (hunk 5 above)
Comment (alexbakharew) [reply tests/js/common/aql/aql-view-arangosearch-ddl-combined.js:1589]: It was before my changes

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**